crates/flox-rust-sdk/src/models/environment/mod.rs

```diff
@@ -358,24 +343,38 @@ pub fn lock_manifest(
     pkgdb: &Path,
     manifest_path: &Path,
     existing_lockfile_path: Option<&Path>,
+    global_manifest_path: &Path,
 ) -> Result<serde_json::Value, EnvironmentError2> {
     let canonical_manifest_path = manifest_path
         .canonicalize()
         .map_err(EnvironmentError2::OpenManifest)?;
+
     let mut pkgdb_cmd = Command::new(pkgdb);
     pkgdb_cmd
-        .args(["manifest", "lock", "--ga-registry"])
-        .arg(canonical_manifest_path);
+        .args(["manifest", "lock"])
+        .arg("--ga-registry")
+        .arg("--global-manifest")
+        .arg(global_manifest_path);
     if let Some(lf_path) = existing_lockfile_path {
         let canonical_lockfile_path = lf_path
             .canonicalize()
             .map_err(EnvironmentError2::BadLockfilePath)?;
-        pkgdb_cmd.arg(canonical_lockfile_path);
+        pkgdb_cmd.arg("--lockfile").arg(canonical_lockfile_path);
```

Comment:
question (non blocking here): do you know if we can already pass lockfile content here too?

that would make building a bit more convenient for managed envs where we dont have direct access to the files.

Reply:
Yeah, you can pass a file path or inline JSON for this flag now I believe